src/vde_switch/datasock.c

```diff
@@ -259,19 +259,24 @@ static void handle_io(unsigned char type,int fd,int revents,void *arg)
 					if (ep != NULL) {
 						mainloop_set_private_data(fd,ep);
 						setup_description(ep,strdup(req->v3.description));
+					} else {
+						remove_fd(fd);
+						return;
 					}
 				}
 				else if(req->v3.version > 2 || req->v3.version == 2) {
 					printlog(LOG_ERR, "Request for a version %d port, which this "
 							"vde_switch doesn't support", req->v3.version);
-					remove_fd(fd); 
-				}
-				else {
+					remove_fd(fd);
+				} else {
```

Comment:
For instance, this is not necessary.

Reply:
I should do something like this?
```
}
else {
````

So this apply to the other if statements too? So from this:
```
if () {
} else {
    remove_fd(fd);
}
```
to this
```
if () {
}
else {
    remove_fd(fd);
}
```

Reply:
Also I've just notice that the `else if` statement on line `252` is in the wrong format, right?

Reply:
Yes, I think it would be better to refactor the project in future with a consistent coding style (for instance the Linux one).

It is not relevant for this pr.

Reply:
Ok, makes sense. A code formatter is made for this :)